### PR TITLE
Add vagrant other schedulers

### PR DESCRIPTION
**Pull request by MorganRodgers**
FYI: Grid engine is in another branch. I was having trouble getting the web node to be able to submit to the head node. My previous solution just used the web node as the sole compute node.

### CODE REVIEW

**Comment by treydock** [comment]
Should try and put things like hosts file and setup scripts that are common in the common directory and symlink like vagrant-with-slurm does for ood-setup.sh. Less to maintain.

File: vagrant-with-pbspro/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Setup
+
+    vagrant plugin install vagrant-vbguest  # only required once, and is not tied to a particular Vagrant file / directory
+    git clone https://github.com/OSC/pbspro-workbench.git
+    cd pbspro-workbench
```
Comment (johrstrom) [inline vagrant-with-pbspro/README.md]: Not sure that this is right, that URL doesn't exist.

File: vagrant-with-pbspro/hosts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
+10.0.0.100  ood
+10.0.0.101  head
+10.0.0.102  pbscompute
```
Comment (johrstrom) [inline vagrant-with-pbspro/hosts]: Both provide `10.0.0.102  pbscompute` but neither have it in their Vagrantfile. Is that a mistake or should there be a third node?

File: vagrant-with-lsf/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+This Vagrant setup installs the version of LSF Community edition that the user provides, but expects version 10.2.0.6. LSF clients are installed to `head` and `ood`. `head` is both the batch and the only compute node.
+
+The LSF quick start: https://www.ibm.com/support/knowledgecenter/en/SSWRJV_10.1.0/lsf_offering/lsfce10.1_quick_start.html
+
+## Setup
+
+    vagrant plugin install vagrant-vbguest  # only required once, and is not tied to a particular Vagrant file / directory
+    git clone https://github.com/MorganRodgers/lsf-workbench.git
+    cd lsf-workbench
```
[the remark this answers is not in the document]
Comment (MorganRodgers) [reply vagrant-with-lsf/README.md]: You're correct. Although in both cases the point of this PR is to centralize the VM definitions into this repo, so I'll want to update those READMEs.

**Comment by treydock** [comment]
This may conflict with #11 for some changes.

**Comment by treydock** [comment]
Merged #11 so there a few merge conflicts to resolve now.

File: vagrant-with-lsf/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    head.vm.provision "shell", path: "head-setup.sh"
+    head.vm.provision "shell", inline: "hostnamectl set-hostname head"
+    head.vm.provision "shell", inline: "cp -f /vagrant/hosts /etc/hosts"
+
+    install_deps!(head)
+    install_lsf!(head)
+  end
+
+  config.vm.define "ood", primary: true, autostart: true do |ood|
+    ood.vm.network "forwarded_port", guest: 80, host: 8080
```
Comment (treydock) [inline vagrant-with-lsf/Vagrantfile]: I'd recommend `auto_correct` or changing port to avoid conflict with other Vagrantfiles.

Comment (treydock) [reply vagrant-with-lsf/Vagrantfile]: Would be `auto_correct: true`

File: vagrant-with-lsf/Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    ood.vm.network "private_network", ip: "10.0.0.100"
```
Comment (treydock) [inline vagrant-with-lsf/Vagrantfile]: Need different IP to avoid conflict with other VMs.

File: vagrant-with-pbspro/Vagrantfile [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+end
+
+Vagrant.configure(2) do |config|
+  config.vm.box = "centos/7"
+  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
+  config.vm.synced_folder "./ood-home", "/home/ood", type: "virtualbox", mount_options: ["uid=1001","gid=1001"]
+
+  config.vm.define "ood", primary: true, autostart: true do |ood|
+    ood.vm.network "forwarded_port", guest: 80, host: 8080
+    ood.vm.network "private_network", ip: "10.0.0.100"
```
Comment (treydock) [inline vagrant-with-pbspro/Vagrantfile]: Need different IP.

File: vagrant-with-pbspro/Vagrantfile [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  SHELL
+end
+
+Vagrant.configure(2) do |config|
+  config.vm.box = "centos/7"
+  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
+  config.vm.synced_folder "./ood-home", "/home/ood", type: "virtualbox", mount_options: ["uid=1001","gid=1001"]
+
+  config.vm.define "ood", primary: true, autostart: true do |ood|
+    ood.vm.network "forwarded_port", guest: 80, host: 8080
```
Comment (treydock) [inline vagrant-with-pbspro/Vagrantfile]: Need `auto_correct: true`.

**Comment by MorganRodgers** [comment]
The port/IP address overlap is intentional at this stage. If we want to allow more than one of these Vagrant 'clusters' to run at once that's a separate job of work because at that point it would make sense to be able to have a single OnDemand instance that can talk to each of these 'clusters'.

**Pull request merged**